source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+
+#. On the controller node, copy the munge key to all nodes and start the 
+   munge service.
+          
+   a. Create the :file:`/etc/munge/` directory on each node.
+
+      .. code-block:: bash
+         
+	 sudo pdsh -P workers mkdir /etc/munge
+   
+   #. Copy the munge key over.
+
+      .. code-block:: bash
+         
+	 sudo pdcp -P workers /etc/munge/munge.key /etc/munge
+
+   #. Set the ownership of the :file:`munge.key` file to `munge`.
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers chown munge: /etc/munge/munge.key
+
+    #. Start the munge service.
+
+      .. code-block:: bash
+
+         sudo pdsh -P workers systemctl enable munge --now
+
+    #. Verify the munge service is running.
+
+       .. code-block:: bash
+
+          sudo pdsh -P workers "systemctl status munge | grep Active"
+      
+       Example output:
+
+       .. code-block:: console
+
+          hpc-worker3:      Active: active (running) since Wed 2020-04-15 19:47:58 UTC; 55s ago
+          hpc-worker4:      Active: active (running) since Wed 2020-04-15 19:47:58 UTC; 55s ago
+          hpc-worker2:      Active: active (running) since Wed 2020-04-15 19:47:59 UTC; 54s ago
+          hpc-worker1:      Active: active (running) since Wed 2020-04-15 19:47:59 UTC; 54s ago
+
+Verify controller can run jobs on all nodes
+*******************************************
+
+#. Check the state of the worker nodes.  
+
+   .. code-block:: bash
+
+      sinfo
+
+   Example output:
+
+   .. code-block:: console
+
+      PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
+      workers*     up   infinite      4   idle hpc-worker[1-4]
+      debug        up   infinite      2   idle hpc-worker[1,3]
+
+   .. tip::
+
+      If the nodes are in a "down" state, put them in the "idle" state.
+
+      .. code-block:: bash
+
+         sudo scontrol update nodename=<worker> state=idle reason=""
+
+#. And finally, verify slurm can run jobs on all 4 worker nodes by issuing
+   a simple :command:`hostname` command.
+
+   .. code-block:: bash
+
+      srun -N4 -p worker hostname
+   
+   Example output:
+     
+   .. code-block:: console
+
+      hpc-worker4
+      hpc-worker3
+      hpc-worker1
+      hpc-worker2
+
+Create and run example scripts
+******************************
+
+Example 1: Return the hostname of each worker and output to :file:`show-hostnames.out`
+======================================================================================
+
+#. On the controller node, create the slurm :file:`show-hostnames.sh` script.
+
+   cat > show-hostnames.sh << EOF
+   #!/bin/bash
+   #
+   #SBATCH --job-name=show-hostnames
+   #SBATCH --output=show-hostnames.out
+   #
+   #SBATCH --ntasks=4
+   #SBATCH --time=10:00
+   #SBATCH --mem-per-cpu=100
+   #SBATCH --ntasks-per-node=1
+
+   srun hostname
+   EOF
+
+#. Execute the script.
+
+   .. code-block:: bash
+
+      sbatch show-hostnames.sh
+   
+   The result will be appear on first node.  
```

Comment:
Redundant  "be", more precisely:
The result will appear on the first node of the used partition. As no partition was explicitly specified, it would be the default partition.

Reply:
I used your version.  Thanks!